hail/python/test/hail/matrixtable/test_matrix_table_from_parts.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import hail as hl
+
+class Inits():
+    globals={'hello': 'world'}
+    rows={'foo': ['a', 'b']}
+    cols={'bar': ['c', 'd']}
+    entries={'baz': [[1, 2], [3, 4]]}
+
+def unless(test: bool, kvs):
+    return {} if test else kvs
+
+def assert_matches_globals(mt: 'hl.MatrixTable', no_props=False):
+    assert hl.eval(mt.globals) == hl.Struct(**unless(no_props, Inits.globals))
+
+def assert_matches_rows(mt: 'hl.MatrixTable', no_props=False):
+    assert mt.rows().order_by('row_idx').collect() == \
```

Comment:
The order_by is unnecessary, the rows table is always keyed by the row key which is, by construction in the method, row_idx.

Reply:
In particular, if a bug was introduced that changed the row key, I'd want the tests to fail.